src/iface/ethernet.rs

```diff
@@ -267,13 +267,16 @@ impl<'b, 'c, 'e, DeviceT> Interface<'b, 'c, 'e, DeviceT>
         self.inner.ip_addrs.as_ref()
     }
 
-    /// Update the IP addresses of the interface.
+    /// Replace the set of IP addresses.
     ///
     /// # Panics
     /// This function panics if any of the addresses are not unicast.
-    pub fn update_ip_addrs<F: FnOnce(&mut ManagedSlice<'c, IpCidr>)>(&mut self, f: F) {
-        f(&mut self.inner.ip_addrs);
-        InterfaceInner::check_ip_addrs(&self.inner.ip_addrs)
+    pub fn set_ip_addrs<S>(&mut self, ips: S)
+        where S: Into<ManagedSlice<'c, IpCidr>>
```

Comment:
The purpose of this method is to be able to update IP addresses without assigning a different ManagedSlice. It's for memory-constrained devices without an allocator.

Reply:
Well you still can't append to it. Should I just make a way to set an ip based on index?